picard/tagger.py

```diff
@@ -346,105 +363,152 @@ def move_files(self, files, target):
             self.cluster(files)
 
     def add_files(self, filenames, target=None):
+        self._add_files([self._add_files_from_directory("", filenames)], target=target)
+
+    def _add_files(self, list_of_filenames, target=None):
         """Add files to the tagger."""
-        ignoreregex = None
-        pattern = config.setting['ignore_regex']
-        if pattern:
-            ignoreregex = re.compile(pattern)
-        ignore_hidden = config.setting["ignore_hidden_files"]
-        new_files = []
+        number_of_files = sum([len(x) for x in list_of_filenames])
+        number_of_directories = len(list_of_filenames)
+        log.debug("Adding a total of %d files in %d directories",
+                number_of_files, number_of_directories)
+        self.window.set_statusbar_message(
+            N_("Adding a total of %(files)d files in %(directories)d directories ..."),
+            {'files': number_of_files, 'directories': number_of_directories},
+            echo=None
+        )
+        for filenames in list_of_filenames:
```

Comment:
I wasn't sure what to call it. 

"filename" would typically refer to string, and "filenames" would refer to a list of "filename". This is a list of filenames, where each member is a list of files in a single directory. 

Similarly "directory" suggests a string containing a directory path. 

That said `for directory in list_of_filenames:` would at first sight seem a bit odd, but you could probably work it out.

How about `filenames_in_directory`?